src/main/java/org/jabref/logic/importer/ImportFormatReader.java

```diff
@@ -215,4 +218,39 @@ public UnknownFormatImport importUnknownFormat(Path filePath) throws ImportExcep
 
         throw new ImportException(Localization.lang("Could not find a suitable import format."));
     }
+
+    public UnknownFormatImport importUnknownFormatFromString(String data) throws ImportException {
```

Comment:
As far as I can see, this is mostly a copy of `importUnknownFormat(Path filePath)` and I would like to have a solution without code duplication. Maybe something along the following lines works?
````java
private UnknownFormatImport importUnknownFormat(Function<Importer, ParserResult> importDatabase, Function<Importer, Boolean> isRecognizedFormat) {
    // original method with obvious changes imFo.importDatabase -> importDatabase.apply(imFo)
}

public UnknownFormatImport importUnknownFormat(Path filePath) {
    return importUnknownFormat(importer -> importer.importDatabase(filePath, importFormatPreferences.getEncoding(), importer -> importer .isRecognizedFormat(filePath, importFormatPreferences.getEncoding());
}

public UnknownFormatImport importUnknownFormat(String data) {
    return importUnknownFormat(importer -> importer.importDatabase(data), importer -> importer .isRecognizedFormat(data));
}
````